keyboards/afternoonlabs/southern_breeze/keymaps/via/keymap.c

```diff
@@ -0,0 +1,73 @@
+                    _______,           _______, _______, _______, _______, _______, _______,                            _______, _______, _______, _______, _______, _______, 
+        //├────────┼────────┼────────┤├────────┼────────┼────────┼────────┼────────┼────────┤                          ├────────┼────────┼────────┼────────┼────────┼────────┤
+           _______, _______, _______,  _______, _______, _______, _______, _______, _______,                            _______, _______, _______, _______, _______, _______,
+        //└────────┴────────┴────────┘└────────┴────────┼────────┼────────┼────────┼────────┼────────┐        ┌────────┼────────┼────────┼────────┴────────┴────────┴────────┘
+                                                                  _______, _______, _______, _______,          _______, _______, _______, _______
+        //                                                       └────────┴────────┴────────┴────────┘        └────────┴────────┴────────┴────────┘
+    ),                             
+                                   
+    [_MAIN] = LAYOUT(
+        //┌────────┬────────┬────────┐┌────────┬────────┬────────┬────────┬────────┬────────┐                          ┌────────┬────────┬────────┬────────┬────────┬────────┐
+           KC_MINS, KC_EQL,  KC_GRV,   KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                               KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, 
+        //├────────┼────────┼────────┤├────────┼────────┼────────┼────────┼────────┼────────┤                          ├────────┼────────┼────────┼────────┼────────┼────────┤
+           KC_LBRC, KC_RBRC, KC_BSLS,  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                               KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  
+        //├────────┼────────┼────────┤├────────┼────────┼────────┼────────┼────────┼────────┤                          ├────────┼────────┼────────┼────────┼────────┼────────┤
+                    KC_UP,             KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                               KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, 
+        //├────────┼────────┼────────┤├────────┼────────┼────────┼────────┼────────┼────────┤                          ├────────┼────────┼────────┼────────┼────────┼────────┤
+           KC_LEFT, KC_DOWN, KC_RIGHT, KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                               KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, 
+        //└────────┴────────┴────────┘└────────┴────────┼────────┼────────┼────────┼────────┼────────┐        ┌────────┼────────┼────────┼────────┴────────┴────────┴────────┘
+                                                                  KC_LCTL, KC_LALT, KC_LGUI, KC_SPC,           KC_ENT,  RAISE,   LOWER,   XXXXXXX
+        //                                                       └────────┴────────┴────────┴────────┘        └────────┴────────┴────────┴────────┘
+    )
+
```

Comment:
Need one more empty layer here.

Reply:
Out of curiosity, why is an extra empty layer required here? Is it just an artifact of how VIA works, or is it a convention that QMK likes to follow?

Reply:
VIA requires a constant number of layers in the keymap, which must match `DYNAMIC_KEYMAP_LAYER_COUNT` (default 4). This is because, when the firmware starts up for the first time, or the EEPROM is reset (same thing, really), the VIA code will retrieve the keymap from flash and place it into EEPROM, where it can be rewritten by the app. If that section of flash contains less keymap data than it is expecting, it will overrun and read garbage; whatever data or code is next to the keymap array. So it is good practice to fill in every layer with `KC_TRNS` in VIA keymaps, even if they are not initially accessible.

Reply:
Awesome - makes sense! Thanks for the detailed answer!